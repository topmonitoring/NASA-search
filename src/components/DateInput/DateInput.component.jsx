import React from "react";
import DatePicker from "react-datepicker";

import { connect } from "react-redux";
import { setCurrentCard, setRandomCard } from "../../redux/card/card.actions";
import { selectCardDate } from "../../redux/card/card.selectors";

import "react-datepicker/dist/react-datepicker.css";
import { SearchBar, CustomButton } from "./Dateinput.styles";

export const DateInput = ({ date, setRandomCard, setCurrentCard }) => (
  <SearchBar>
    <span>
      Select a Date:
      <DatePicker selected={date} onChange={date => setCurrentCard(date)} />
    </span>{" "}
    <CustomButton onClick={() => setRandomCard()}>
      Generate Random photo
    </CustomButton>
  </SearchBar>
);
const mapStateToProps = state => ({
  date: selectCardDate(state)
});

const mapDispatchToProps = despatch => ({
  setCurrentCard: date => despatch(setCurrentCard(date)),
  setRandomCard: () => despatch(setRandomCard())
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DateInput);
