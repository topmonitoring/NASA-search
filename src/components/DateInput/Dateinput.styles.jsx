import styled from "styled-components";

export const SearchBar = styled.div`
  width: 60%;
  margin: auto;
  border-radius: 25px;
  background-color: rgb(222, 222, 222);
  font-size: 25px;
  padding: 10px;
  span {
    font-size: 18px;
  }
`;
export const CustomButton = styled.button`
  height: 25px;
`;
