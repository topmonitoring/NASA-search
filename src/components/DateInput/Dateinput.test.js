import React from "react";
import { shallow } from "enzyme";
import { DateInput } from "./DateInput.component";

let wrapper;
beforeEach(() => {
  const mockProps = {
    date: "22-11-2018",
    setRandomCard: jest.fn(),
    setCurrentCard: jest.fn()
  };
  wrapper = shallow(<DateInput store={mockProps} />);
});

it("renders DateInput component", () => {
  expect(wrapper).toMatchSnapshot();
});
