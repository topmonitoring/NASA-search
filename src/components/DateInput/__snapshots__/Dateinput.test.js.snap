// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DateInput component 1`] = `ShallowWrapper {}`;
